packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -524,6 +558,35 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		entry.handleP.resolve(this.getBlobHandle(localId));
 	}
 
+	private sendBlobAttachOnResubmit(localId: string, blobId: string): void {
+		console.log("snedBlobAttachOnResubmit");
+		const entry = this.pendingBlobs.get(localId);
+		if (!entry) {
+			console.log("NO ENTRY");
+			return this.sendBlobAttachOp(localId, blobId);
+		}
+		if (entry.minTTLInSeconds && entry.serverUploadTime) {
+			console.log("Entry has minTTL");
+			const timeLapseSinceServerUpload = (Date.now() - entry.serverUploadTime) / 1000;
+			console.log(timeLapseSinceServerUpload, " ", entry.minTTLInSeconds);
+			if (entry.minTTLInSeconds - timeLapseSinceServerUpload < 0) {
+				console.log("expired");
+				this.uploadExpiredBlob(localId, entry.blob).then(
```

Comment:
If this is an offline blob, there may be ops behind this BlobAttach op that reference the blob handle. Since we can't await here, the ops will be submitted and these handles will not resolve for remote clients until this BlobAttach op goes through (they will error and close I think), and if the client disconnects before submitting it, or the upload fails, the handle will be permanently broken.